### PR TITLE
Andreyan/compress weights docs

### PR DESCRIPTION
**Pull request by andreyanufr**
### Changes

Added description of new weights compression functionality.

### Related tickets

CVS-115948

### Tests

No test.

### CODE REVIEW

File: docs/compression_algorithms/CompressWeights.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+### Compress Weights
+
+#### The algorithm description
+
+The Weights Compression algorithm compresses weights in LLM to int8 and adds extra decompression layers to the input network or keeps original weights precision and inserts FakeQuantize operations. The goal is to compress weights of big LLMs before conversion to IR without usage of dataset. The algorithm compresses weights only for Linear and Embedding layers.
+
+#### User guide
+
```
Comment (MaximProshin) [inline docs/compression_algorithms/CompressWeights.md:8]: Can you add comment before each example about the difference?

File: docs/compression_algorithms/CompressWeights.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+### Compress Weights
+
+#### The algorithm description
+
+The Weights Compression algorithm compresses weights in LLM to int8 and adds extra decompression layers to the input network or keeps original weights precision and inserts FakeQuantize operations. The goal is to compress weights of big LLMs before conversion to IR without usage of dataset. The algorithm compresses weights only for Linear and Embedding layers.
```
Comment (MaximProshin) [inline docs/compression_algorithms/CompressWeights.md:5]: Make clear what is default

File: docs/compression_algorithms/CompressWeights.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+### Compress Weights
```
Comment (AlexKoff88) [inline docs/compression_algorithms/CompressWeights.md:1]:
```suggestion
### Weights Compression
```

File: docs/compression_algorithms/CompressWeights.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+#### The algorithm description
+
+The Weights Compression algorithm compresses weights in LLM to int8 and adds extra decompression layers to the input network. The goal is to compress weights of big LLMs before conversion to IR without usage of dataset. The algorithm compresses weights only for Linear and Embedding layers. It is also possible to keeps original weights precision and inserts FakeQuantize operations by setting use_fake_quantize parameter to True.
```
Comment (AlexKoff88) [inline docs/compression_algorithms/CompressWeights.md:5]:
```suggestion
The Weights Compression algorithm is aimed at compressing the weights of the models and can be used to optimize the model footprint and performance of large models where the size of weights is relatively larger than the size of activations, for example, Large Language Models (LLM). The algorithm compresses weights only for Linear and Embedding layers. It is also possible to keep the precision of the original weights and insert FakeQuantize operations by setting `use_fake_quantize` parameter to True.

compresses weights in LLM to int8 and adds extra decompression layers to the input network. The goal is to compress weights of big LLMs before conversion to IR without usage of dataset. The algorithm compresses weights only for Linear and Embedding layers. It is also possible to keeps original weights precision and inserts FakeQuantize operations by setting use_fake_quantize parameter to True.
```

**Review by AlexKoff88 — APPROVED** [review]

**Pull request merged**